druid/src/widget/images.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An Image widget.
+//! Please consider using SVG and the SVG wideget as it scales much better.
+
+use std::error::Error;
+use std::marker::PhantomData;
+use std::str::FromStr;
+
+use image;
+
+use crate::{
+    Affine, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+    PaintCtx, Point, Rect, RenderContext, Size, UpdateCtx, Widget,
+};
+
+use crate::piet::{ImageFormat, InterpolationMode};
+
+#[derive(PartialEq)]
+pub enum FillStrat {
```

Comment:
This will need documentation; it might be easiest to link to some existing resource for this? We should also document each member. 😢

also it's probably worth deriving at least `Clone, Copy` on this.

Reply:
I have linked to flutter which some one sent me on zulip. the link documents each one en detail and i think i have implemented them 1:1 for all.

Not sure if what i have done is enough or if i should reiterate the salient points in our code again.